V3 "Fire in the Hole"

trueter opened the pull request:
https://github.com/membranepotential/ascstudymonitor/milestone/2

trueter committed:
social sharing (#11)

Closes: #11

trueter committed:
skip cypress install on ci

trueter committed:
direct link to entry button #33

trueter committed:
always collect coverage on ci

trueter committed:
keyword filtering and navigation limit to 10

trueter committed:
add product analytics

trueter committed:
initial mobile support

trueter committed:
refactor state management, mock recommendations, break most tests

trueter committed:
fix math

19h commented on client/public/index.html:
```diff
@@ -93,5 +93,70 @@
+              'once',
+              'off',
+              'on',
+              'addSourceMiddleware',
+              'addIntegrationMiddleware',
+              'setAnonymousId',
+              'addDestinationMiddleware',
+            ]
+            analytics.factory = function(e) {
+              return function() {
+                var t = Array.prototype.slice.call(arguments)
+                t.unshift(e)
+                analytics.push(t)
+                return analytics
+              }
+            }
+            for (var e = 0; e < analytics.methods.length; e++) {
+              var t = analytics.methods[e]
+              analytics[t] = analytics.factory(t)
+            }
+            analytics.load = function(e, t) {
+              var n = document.createElement('script')
+              n.type = 'text/javascript'
+              n.async = !0
+              n.src =
+                'https://cdn.segment.com/analytics.js/v1/' +
+                e +
+                '/analytics.min.js'
+              var a = document.getElementsByTagName('script')[0]
+              a.parentNode.insertBefore(n, a)
+              analytics._loadOptions = t
+            }
+            analytics.SNIPPET_VERSION = '4.1.0'
+            analytics.load(
+              window.ENV === 'production'
+                ? '8OvSbZ17xxOC5FV3RkyM0bGSPVjBCkA6'
+                : 'aN2mtGYzr15b3W6HnNLfgXIsmPnBdnwt',
+            )
+          }
+      })()
```
please tell me this is the result of beautifying a minified tracking snippet

trueter replied:
It is.

trueter committed:
fix menu

softrrrat committed:
remove e2e tests until cypress installs properly

softrrrat committed:
Update .gitignore

softrrrat committed:
- XD File Version 1

softrrrat committed:
- update XD file with Search Takeover

trueter committed:
remove root/package.json

trueter committed:
push partial progress on v3 redesign

softrrrat committed:
- update XD file

trueter committed:
fix font, add hotkey, initial search ui

trueter committed:
progress on search ui

softrrrat committed:
Update .gitignore

softrrrat committed:
Update asc_redesign_2_1_A.xd

trueter committed:
add initial foxglove service for thumbnail rendering

membranepotential committed:
Restructure backend subrepository

trueter committed:
rework list view, some config, update packages for node 14

trueter committed:
ui work on rows and search button

trueter committed:
clean up ui

trueter committed:
clean up and fix tests

trueter committed:
use waypoint to show additional search bar

trueter committed:
work on search modal and ui refactors

trueter committed:
refactor and test fixes

trueter committed:
update dependencies

trueter committed:
remove unused env vars from github action

trueter committed:
initial work on publication single view

trueter committed:
fix and clean up tests

trueter committed:
remove coveralls for now

membranepotential committed:
Provisional GraphQL Schema

membranepotential committed:
GraphQL Schema: Everything camelCase

membranepotential committed:
Update schema with response types and documentation

membranepotential committed:
Rebuild backend first pass

membranepotential committed:
Add stub for recommendations

membranepotential committed:
Rename document -> publication

membranepotential committed:
Rename source -> journal

membranepotential committed:
Implement document based event log

membranepotential committed:
Add mongodb based lock for update

membranepotential committed:
Change terminology document -> publication

membranepotential committed:
Implement search engine

membranepotential committed:
Tutti Frutti Backend Commit

 - document -> publication in more places
 - kicked out whoosh in favour of mongodb text search
 - search queries and filtering implemented, but not yet sorted by
   relevance
 - refined schema after consultation with frontend department
 - implemented filterable fields as GraphQL types with publication
   counts
 - get total count of publications for a search query
 - more tests as filter behaviour specification

membranepotential committed:
Implement filterable field distinct aggregation queries

membranepotential committed:
Switch backend to fancy new ASGI tech :nerd_face:

membranepotential committed:
Implement field suggestions via ngram store

membranepotential committed:
Lowercase before generating ngrams

membranepotential committed:
Fix embedding of filterable array fields

membranepotential committed:
Serve index.html with metatags by backend even in prod

membranepotential committed:
Test ngram store

membranepotential committed:
Implement post queue management

membranepotential committed:
More correct lock handling in post queue

trueter committed:
initial graphql commit, breaks things! => need to fix cors

trueter commented:
@19h graphql stuff getting hot here. reviews welcome :*

membranepotential committed:
Wide open CORS

membranepotential committed:
Draft for publication list, next up: pagination

membranepotential committed:
Move schema.graphql to repo root

trueter committed:
fix favicon

trueter committed:
fix disciplines warning

trueter committed:
cleanup

trueter committed:
allow importing of .gql files

trueter committed:
progress on graphql search in modal

trueter committed:
initial implementation of primary search

trueter committed:
allow possible types def to combat heuristic fragment matching

trueter committed:
axe vue-tailwind

trueter committed:
remove apollo composable for now

trueter committed:
implement infinite scroll on list

trueter committed:
handle publications without journal according to schema

membranepotential committed:
Draft for queue management client

trueter committed:
patch issue of queue items being returned by graphql in wrong order

trueter committed:
refactors & initial publication detail view

membranepotential committed:
Backend improvements

 - Publication is typesafe
 - Added search result pagination
 - Fixed queue order

membranepotential committed:
Various fixes for backend

membranepotential committed:
Fix author camelCase for fieldSuggestions

membranepotential committed:
Fix author camelCase in other resolvers

membranepotential committed:
Stub field suggestions with filters

membranepotential committed:
Implement post mutation

trueter committed:
initial filter implementation in search

membranepotential committed:
Ignore empty lists in graphql filters

membranepotential committed:
Setup infrastructure for staging

membranepotential committed:
Get nginx routing right

membranepotential committed:
Implement prefix matching in ngram store

membranepotential committed:
Test poster

membranepotential committed:
Little backend fixes

membranepotential committed:
Initial draft for single publication view

trueter committed:
progress on single view, clean up and modal love

membranepotential committed:
Experiments with workflows

membranepotential committed:
Backend serves static files in development again

trueter committed:
remove legacy files

trueter committed:
search love

trueter committed:
activate codeql analysis

https://github.blog/2020-09-30-code-scanning-is-now-available/

membranepotential committed:
Add script that posts

membranepotential committed:
Backend fixes

membranepotential committed:
Make client tests run

membranepotential committed:
Implement simple title search

membranepotential committed:
Client lint fix

membranepotential committed:
Bubbles fixed

softrrrat committed:
- search bar adjusments

softrrrat committed:
Merge branch 'v2.1' of https://github.com/membranepotential/ascstudymonitor into v2.1

trueter committed:
much desired frontend cleanup

membranepotential committed:
Fix total publication count

membranepotential committed:
Fix totalCount is 0

trueter committed:
enable viewing search results

membranepotential committed:
Backend: Filters & Tests

 - FilterList as dataclass
 - Exclude filters in ngram search
 - Test graphql endpoints

trueter committed:
remove legacy python / js data bridge

trueter committed:
add empty view maessage to list

trueter committed:
hardcore publication path /p/:slug

trueter committed:
persist search in query string, navigtate to single pub from search

trueter committed:
example of adding a filter from the pub list items / detail view

membranepotential committed:
Integrate test and deploy actions

membranepotential committed:
Disable unused variable lint error on unused rest sibling

membranepotential committed:
Better make rule for stage deploy

membranepotential committed:
Add scroll lock when search modal is open

trueter committed:
fix eslint

trueter commented on client/package.json:
```diff
@@ -68,6 +68,16 @@
+        {
+          "vars": "all",
+          "args": "after-used",
+          "ignoreRestSiblings": true
```
👍

trueter committed:
support more filter actions, ui cleanup

trueter committed:
Merge branch 'master' into v2.1

trueter committed:
add back button to filtered list views

trueter committed:
fix tests

trueter committed:
update deps

trueter committed:
get rid off less

trueter committed:
cleanup, restore fade animation

trueter committed:
progress on queue manager interface. some mutations work, some don’t 👼

membranepotential committed:
Test PublicationStore with FilterList

trueter committed:
wrap up queue manager

pablopaul committed:
Fix poetry config command

trueter commented:
Closing to clean things up. We’re now working on the `next` branch.
